notebooks/generative-ai/question-answering.ipynb

```diff
@@ -88,34 +90,44 @@
         "\n",
         "embeddings = OpenAIEmbeddings(openai_api_key=OPENAI_API_KEY)\n",
         "\n",
-        "vector_store = ElasticsearchStore(es_cloud_id=CLOUD_ID, es_user=CLOUD_USERNAME, es_password=CLOUD_PASSWORD,\n",
-        "            index_name= \"workplace_index\",\n",
-        "            embedding=embeddings\n",
-        "        )\n"
+        "vector_store = ElasticsearchStore(\n",
+        "    es_cloud_id=CLOUD_ID,\n",
+        "    es_user=CLOUD_USERNAME,\n",
+        "    es_password=CLOUD_PASSWORD,\n",
+        "    index_name= \"workplace_index\",\n",
+        "    embedding=embeddings\n",
+        ")\n"
       ]
     },
     {
       "cell_type": "markdown",
       "metadata": {},
       "source": [
-        "## Download the dataset \n",
+        "## Indexing Data into Elasticsearch\n",
         "\n",
         "Let's download the sample dataset and deserialize the document."
       ]
     },
     {
       "cell_type": "code",
-      "execution_count": null,
+      "execution_count": 55,
       "metadata": {
         "id": "J8-93TiJsNyK"
       },
       "outputs": [],
       "source": [
+        "from urllib.request import urlopen\n",
+        "from langchain.text_splitter import CharacterTextSplitter\n",
+        "from langchain.llms import OpenAI\n",
+        "import json\n",
+        "\n",
         "url = \"https://raw.githubusercontent.com/elastic/elasticsearch-labs/main/example-apps/workplace-search/data/data.json\"\n",
         "\n",
         "response = urlopen(url)\n",
+        "data = json.load(response)\n",
         "\n",
-        "workplace_docs = json.loads(response.read())\n"
+        "with open('data.json', 'w') as json_file:\n",
+        "    json.dump(data, json_file)\n"
```

Comment:
Seems strange that you are downloading this file, just to save it again, decode and re-encoding the JSON. Doesn't look like there is a JSON URL downloader in langchain though, so maybe this is the best that can be done.

Reply:
Yeh its not great but unfortunately it doesn't support loading from url or from a blob so its the best I can do. I prefer doing it via loader than manually as its easier to read and relate to their docs.